### ISSUE TITLE
nill pointer on PUT with no body

### ISSUE THREAD

**Issue by bnfbiz** [flaccid/j2xrp#2]
Received the following error when doing a **PUT** request that doesn't need any XML in the body:

```
time="2018-09-15T18:02:52Z" level=info msg="starting http server"
time="2018-09-15T18:02:52Z" level=info msg="proxy requests to https://<url replaced>"
time="2018-09-15T18:02:52Z" level=info msg="listening for requests on :80"
2018/09/15 18:02:55 http: panic serving 10.0.2.4:56236: runtime error: invalid memory address or nil pointer dereference
goroutine 5 [running]:
net/http.(*conn).serve.func1(0xc420090960)
/usr/lib/golang/src/net/http/server.go:1697 +0xd0
panic(0x79ab60, 0xa01d40)
/usr/lib/golang/src/runtime/panic.go:491 +0x283
bytes.(*Buffer).ReadFrom(0xc42003ba68, 0x0, 0x0, 0x0, 0x0, 0xffffffffffffffff)
/usr/lib/golang/src/bytes/buffer.go:209 +0x142
github.com/flaccid/j2xrp/proxy.Serve.func1(0xc42010e100)
/root/go/src/github.com/flaccid/j2xrp/proxy/proxy.go:87 +0x187
net/http/httputil.(*ReverseProxy).ServeHTTP(0xc420051780, 0x9d6040, 0xc42008c1c0, 0xc42010e000)
/usr/lib/golang/src/net/http/httputil/reverseproxy.go:169 +0x17c
net/http.serverHandler.ServeHTTP(0xc4200f0410, 0x9d6040, 0xc42008c1c0, 0xc42010e000)
/usr/lib/golang/src/net/http/server.go:2619 +0xb4
net/http.(*conn).serve(0xc420090960, 0x9d6900, 0xc420051880)
/usr/lib/golang/src/net/http/server.go:1801 +0x71d
created by net/http.(*Server).Serve
/usr/lib/golang/src/net/http/server.go:2720 +0x288

```

**Comment by flaccid** [comment]
@bnfbiz pushed a fix, maybe that is good enough? Can you test it out please and if its ok, I'll update the docker image.

**Comment by bnfbiz** [comment]
@flaccid I tested the fix and it appears to work.  I also reviewed it (I was working on a similar fix), and see that the fix created basically an empty xml body.  I don't know, but could that cause issues to a REST API that didn't need any payload?  In my tests it worked fine, just wasn't sure if it would work in all cases.  My code was leaving the body empty.

EIther way the fix that was proposed is working for me.
